Create 00-trident.md (#1389)
spectrocloud/librarium: content/docs/06-integrations/00-trident.md

# Versions Supported

<Tabs>

<Tabs.TabPane tab="23.01.x" key="trident_23.01">

## Prerequisites

- All worker nodes in the Kubernetes cluster must be able to mount the volumes you have provisioned for your pods. To prepare the worker nodes, you must install NFS or iSCSI tools based on your driver selection. The docs on selecting the right tools, can be found [here](https://docs.netapp.com/us-en/trident/trident-use/worker-node-prep.html#selecting-the-right-tools).

- Port requirements as listed in the table. 

| Port  | Purpose                                                      |
|-------|--------------------------------------------------------------|
| 8443  | Backchannel HTTPS                                            |
| 8001  | Prometheus metrics endpoint                                  |
| 8000  | Trident REST server                                          |
| 17546 | Liveness/readiness probe port used by Trident daemonset pods |

<InfoBox>

> The liveness/readiness probe port can be changed during installation using the `--probe-port` flag. It is important to make sure this port isn’t being used by another process on the worker nodes. 

Reviewer: ```suggestion
The liveness and readiness probe port can be changed during installation using the `--probe-port` flag. Ensure the probe port is not used by another process on the worker nodes. 
```

Active voice changes.

</InfoBox>

## Parameters

> | Parameter | Description |

Reviewer: Let's add another column called "Default Values" and provide each default into its own respective cell. That makes it easier for the reader to scan the table.

| ``tridentLogFormat`` | Sets the Trident logging format (text or json). Default: `text` |
| ``tridentDisableAuditLog`` | Disables Trident's audit logger. Default: `true` |
| ``tridentImage`` | Allows the complete override of the image for Trident. |
| ``tridentImageTag`` | Allows overriding the tag of the image for Trident. |
| ``tridentEnableNodePrep`` | (Deprecated) Attempts to automatically install required packages on nodes. Default: `false` |
| ``tridentSkipK8sVersionCheck`` | (Deprecated) Allows overriding the k8s version limit for Trident. Default: `false` |
| ``tridentProbePort`` | Allows overriding the default port used for k8s liveness/readiness probes. |
| ``windows`` | Allows Trident to be installed on Windows worker node. Default: `false` |
| ``enableForceDetach`` | Allows enabling the force detach feature. Default: `false` |
| ``excludePodSecurityPolicy`` | Excludes the operator pod security policy from creation. Default: `false` |

## Usage

After deploying Trident, you will need to create a backend and a storage class before you can start provisioning volumes and mounting those to any pods. 

<br />

> #### Create a backend.  

Reviewer: ```suggestion
#### Create a Storage Backend  
```

> Trident supports multiple backends and depending your storage backend, you will need to choose a backend that fits your needs. You can find example driver manifests [here](https://github.com/NetApp/trident/tree/master/trident-installer/sample-input/backends-samples) and you will need to personalise them with your credentials and environment configurations. Here's an example with the ONTAP-NAS driver. 

Reviewer: ```suggestion
Trident supports multiple storage backends. Select a supported backend that fits your needs. You can find example of different backends by reviewing the driver manifests examples in the official Trident [repository]((https://github.com/NetApp/trident/tree/master/trident-installer/sample-input/backends-samples). If you decide to use one of the example configurations, make sure you update the configuration with your credentials and environment configurations. Here's an example with the ONTAP-NAS driver. 
```
Language cleanup and clarification

<br />

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: backend-ontap-nas-secret
  namespace: trident
type: Opaque
stringData:
  username: [USERNAME]
  password: [PASSWORD]
---
apiVersion: trident.netapp.io/v1
kind: TridentBackendConfig
metadata:
  name: backend-ontap-nas
  namespace: trident
spec:
  version: 1
  storageDriverName: ontap-nas
  managementLIF: [ONTAP_MANAGEMENT_LIF_IP]
  dataLIF: [DATA_LIF_IP]
  backendName: ontap-nas
  autoExportCIDRs:
  - [x.x.x.x/xx]
  autoExportPolicy: true
  svm: [SVM]
  credentials: 
    name: backend-ontap-nas-secret
```

<br />

> #### Create a storage class.  

Reviewer: ```suggestion
#### Create a Storage Class 
```

> Kubernetes allows us to bind statically and/or dynamically provisioned volumes to our Pods. Statically provisioned volumes are manually created by a user and then referenced in a deployment. Astra Trident allows you to leverage your NetApp storage and 
> Before you can request dynamically provisioned volumes, you will need to create a storage class. Storage classes 

Reviewer: ```suggestion
Kubernetes supports the ability to bind statically or dynamically provisioned volumes to Pods. Statically provisioned volumes are manually created by a user and then referenced in a deployment. Astra Trident allows you to leverage your NetApp storage. You must create a storage class before you can request dynamically provisioned volumes. Storage classes 
```

@Sheldonwl This paragraph seems to be missing a sentence related to the "Astra Trident allows you to leverage your NetApp storage" sentence.

parameters:
  backendType: "ontap-nas"
  csi.storage.k8s.io/fstype: ext4
```

<br />

#### Provision your first Volume.  

Here's an example of a basic Persistent Volume Claim (PVC). 

<br />

```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: trident-pvc
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
  storageClassName: trident-csi
```

<br />

#### Deploy your first Pod and mount the volume.  

Here's an example of a simple Pod deployment. This Pod contains an NGINX container, with a claim to the previously created Persistent Volume Claim (PVC). Once deployed, the PVC with it's bound Persistent Volume (PV), will mount to the Pod and provide persistent storage to this application.    

<br />

```yaml
kind: Pod
apiVersion: v1
metadata:
  name: example-pv-pod
  namespace: trident
spec:
  volumes:
    - name: example-pv-vol
      persistentVolumeClaim:
       claimName: trident-pvc
  containers:
    - name: example-pv-pod
      image: nginx
      ports:
        - containerPort: 80
          name: "http-server"
      volumeMounts:
        - mountPath: "/usr/share/nginx/html"
          name: example-pv-vol

> ```

Reviewer: Let's merge these into one heading called  ### Example Deployment.  In the intro, explain what the example configurations will do. You already have the text, it just needs to be massaged into an intro for a merged section.

<br />

</Tabs.TabPane>

</Tabs>

> # References
> - [NetApp Astra Trident Docs](https://docs.netapp.com/us-en/trident/index.html)
> - [Necessary tools](https://docs.netapp.com/us-en/trident/trident-use/worker-node-prep.html#selecting-the-right-tools)
> - [Trident Github](https://github.com/NetApp/trident)

Reviewer: ```suggestion
# References

- [NetApp Astra Trident Docs](https://docs.netapp.com/us-en/trident/index.html)

- [Necessary tools](https://docs.netapp.com/us-en/trident/trident-use/worker-node-prep.html#selecting-the-right-tools)

- [Trident Github](https://github.com/NetApp/trident)

<br />
```

Added spacing to improve layout.
